BlakeDonn/ideabox-boilerplate
# Final Functionality & Refactor

#### What’s this PR do? 
1. This pull requests adds functionality to the showStarredButton and showAllButton.

2. Created a modal to adjust our nav to better display in mobile, when the user clicks on the hamburger, they are presented with the modal containing the showStarredButton and showAllButton.

3. Refactored our eventListener for "keyup" to contain an anonymous function that invoked the relevant function. This will either provide functionality to the search ideas input, or disable and the saveIdea button accordingly.

4. Adjusted our HTML, CSS, and JavaScript files to follow the Turing style guides.

#### Where should the reviewer start? 

* Start at the HTML and check out that it correctly follows the styling guide, make sure new modal tags are correct.

* Check CSS file to ensure it follows the styling guide.

* Check the JavaScript files and make sure data model interaction is being done correctly for the modal adjustments.

#### How should this be manually tested?

* Test functionality of the filter buttons, and that they accurately hide/show the favorited stars accordingly, check mobile view for modal. Check that search input is accurately updating the displayed cards based on the search, and that it displays correctly when backspacing or deleting.

####  Any background context you want to provide?

* Some refactoring could still be done on the JS, for now it does provide a bug free execution of our application.

#### Screenshots (if appropriate):
